tests/interfaces/messages/handlers/messages-get.spec.ts

```diff
@@ -0,0 +1,240 @@
+    expect(reply.entries).to.not.exist;
+    expect(reply.data).to.not.exist;
+  });
+
+  it('returns a 400 if message contains an invalid message cid', async () => {
+    const alice = await DidKeyResolver.generate();
+    const { recordsWrite } = await TestDataGenerator.generateRecordsWrite({ requester: alice });
+
+    const { message } = await TestDataGenerator.generateMessagesGet({
+      requester   : alice,
+      messageCids : [await Message.getCid(recordsWrite.message)]
+    });
+
+    message.descriptor.messageCids = ['hehetroll'];
+
+    const reply: MessagesGetReply = await dwn.processMessage(alice.did, message);
+
+    expect(reply.status.code).to.equal(400);
+    expect(reply.status.detail).to.include('is not a valid CID');
+    expect(reply.messages).to.be.undefined;
+  });
+
+  it('returns all requested messages', async () => {
+    const did = await DidKeyResolver.generate();
+    const alice = await TestDataGenerator.generatePersona(did);
+    const messageCids: string[] = [];
+
+    const { recordsWrite, dataStream } = await TestDataGenerator.generateRecordsWrite({
+      requester: alice
+    });
+
+    let messageCid = await MessagesGet.getCid(recordsWrite.message);
+    messageCids.push(messageCid);
+
+    let reply = await dwn.processMessage(alice.did, recordsWrite.toJSON(), dataStream);
+    expect(reply.status.code).to.equal(202);
+
+    const { recordsDelete } = await TestDataGenerator.generateRecordsDelete({
+      requester : alice,
+      recordId  : recordsWrite.message.recordId
+    });
+
+    messageCid = await Message.getCid(recordsDelete.message);
+    messageCids.push(messageCid);
+
+    reply = await dwn.processMessage(alice.did, recordsDelete.toJSON());
+    expect(reply.status.code).to.equal(202);
+
+    const { protocolsConfigure } = await TestDataGenerator.generateProtocolsConfigure({
+      requester: alice
+    });
+
+    messageCid = await Message.getCid(protocolsConfigure.message);
+    messageCids.push(messageCid);
+
+    reply = await dwn.processMessage(alice.did, protocolsConfigure.toJSON());
+    expect(reply.status.code).to.equal(202);
+
+    const { messagesGet } = await TestDataGenerator.generateMessagesGet({
+      requester: alice,
+      messageCids
+    });
+
+    const messagesGetReply: MessagesGetReply = await dwn.processMessage(alice.did, messagesGet.toJSON());
+    expect(messagesGetReply.status.code).to.equal(200);
+    expect(messagesGetReply.messages!.length).to.equal(messageCids.length);
+
+    for (const messageReply of messagesGetReply.messages!) {
+      expect(messageReply.messageCid).to.not.be.undefined;
+      expect(messageReply.message).to.not.be.undefined;
+      expect(messageCids).to.include(messageReply.messageCid);
+
+      const cid = await Message.getCid(messageReply.message!);
+      expect(messageReply.messageCid).to.equal(cid);
+    }
+  });
+
+  it('returns message as undefined in reply entry when a messageCid is not found', async () => {
```

Comment:
Sanity: is this meant to be the test that tests the line above that is never hit?

Reply:
na. this test is not supposed to hit the error code path